### PR TITLE
DM-15005: Add a new AngleUnit: milliarcseconds

### PR DESCRIPTION
**Pull request by r-owen**

### CODE REVIEW

File: tests/test_angle.py [hunk 1]
```diff
@@ -202,12 +208,14 @@ def testWrap(self):
...
             ctrAngArcmin = ctrAng.asArcminutes()
             ctrAngArcsec = ctrAng.asArcseconds()
+            ctrAngMilliarcsec = ctrAng.asMilliarcseconds()
             # wrapCtr promises -pi <= ctrAngRad < pi only for radians,
             # but it seems to work for all units
             self.assertGreaterEqual(ctrAngRad, -math.pi)
             self.assertGreaterEqual(ctrAngDeg, -180)
             self.assertGreaterEqual(ctrAngArcmin, -180 * 60)
             self.assertGreaterEqual(ctrAngArcsec, -180 * 3600)
+            self.assertGreaterEqual(ctrAngMilliarcsec, -180 * 3.6e6)
```
Comment (laurenam) [inline tests/test_angle.py]: Any reason you didn't add the **assertLess** for **ctrAngMilliarcsec** below?

File: tests/test_angle.py [hunk 2]
```diff
@@ -178,17 +181,20 @@ def testWrap(self):
...
             self.assertGreaterEqual(posAngArcmin, 0)
             self.assertGreaterEqual(posAngArcsec, 0)
+            self.assertGreaterEqual(posAngMilliarcsec, 0)
             # wrap promises posAng < 2*pi only for radians,
             # but it seems to work for all units
             self.assertLess(posAngRad, 2*math.pi)
             self.assertLess(posAngDeg, 360)
             self.assertLess(posAngArcmin, 360 * 60)
             self.assertLess(posAngArcsec, 360 * 3600)
+            self.assertLess(posAngMilliarcsec, 360 * 3.6e6)
```
Comment (laurenam) [inline tests/test_angle.py]: It still bothers me that you choose to (inconsistently) violate our style guide (https://developer.lsst.io/python/style.html?highlight=whitespace#binary-operators-should-be-surrounded-by-a-single-space-except-for).
Specifically:

> Never surround these binary arithmetic operators with whitespace:
> - multiplication (*),
> - division (/),

File: include/lsst/geom/Angle.h [hunk 3]
```diff
@@ -107,6 +107,10 @@ AngleUnit constexpr degrees = AngleUnit(PI / 180.0);              ///< constant
 AngleUnit constexpr hours = AngleUnit(PI * 15.0 / 180.0);         ///< constant with units of hours
 AngleUnit constexpr arcminutes = AngleUnit(PI / 60 / 180.0);      ///< constant with units of arcminutes
 AngleUnit constexpr arcseconds = AngleUnit(PI / 180.0 / 3600.0);  ///< constant with units of arcseconds
+// Note: if we use PI / 180.0 / 3.6e6 then 60*60*180*1000*lsst.geom.milliarcseconds
+// does not test exactly equal to 180*lsst.geom.degrees
```
Comment (laurenam) [inline include/lsst/geom/Angle.h]: Any comment as to why?  I assume the equivalent works for arcseconds as defined, so is it a precision/roundoff issue?  Is this worrisome??

File: tests/test_angle.py (hunk 1 above)
Comment (r-owen) [reply tests/test_angle.py]: That was an oversight. Thank you for catching it.

File: tests/test_angle.py (hunk 2 above)
Comment (r-owen) [reply tests/test_angle.py]: In this case I have chosen to make the new code look like the old code. I confess that I find lack of spaces hampers readability (except for complex expressions that mix * and / with +-)

File: include/lsst/geom/Angle.h (hunk 3 above)
Comment (r-owen) [reply include/lsst/geom/Angle.h]: I am sure it is a roundoff error, but I am not sure why. The test seems less than ideal; I would have expected `assertAnglesAlmostEqual` in order to handle tiny roundoff errors. But I went with what matches the existing code.

**Pull request merged**